### PR TITLE
[Documentation]: Tables and Extensions BP sections

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
~~WIP~~

I think these might be the last two or 3 that need an initial wave of organizing and setup.

A final round of finishing touches should be done once all these broad-scale organization ones are merged.

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@bendichter This PR is a first-pass polishing of https://nwbinspector--125.org.readthedocs.build/en/125/best_practices/tables.html and https://nwbinspector--125.org.readthedocs.build/en/125/best_practices/extensions.html

File: docs/best_practices/extensions.rst [hunk 1]
```diff
@@ -1,39 +1,49 @@
...
 ==========
 
-Extend only when necessary. Extensions are an essential mechanism to integrate data with NWB that is otherwise not
-supported. However, we here need to consider that there are certain costs associated with extensions, e.g., cost of
-creating, supporting, documenting, and maintaining new extensions and effort for users to use and learn extensions.
-As such, users should attempt to use core neurodata_types or existing extentions before creating extensions.
-``DynamicTables``, which are used throughout the NWB schema e.g., to store information about time intervals and
-electrodes, provide the ability to dynamically add columns without the need for extensions, and can help avoid the
-need for custom extensions in many cases.
+Extend existing NWBFile structures only when absolutely necessary. Extensions are an essential mechanism to integrate
```
Comment (bendichter) [inline docs/best_practices/extensions.rst:4]: I don't like the use of "absolutely" here. It sounds like we are discouraging users from creating extensions

File: docs/best_practices/extensions.rst [hunk 2]
```diff
@@ -1,39 +1,49 @@
...
+data with NWB that is otherwise not supported. However, we here need to consider that there are certain costs associated
+with extensions, *e.g.*, cost of creating, supporting, documenting, and maintaining new extensions and effort for users
+to use and learn already-created extensions.
+
+As such, users should attempt to use core ``neurodata_types`` or pre-existing extentions before creating new ones.
+:nwb-schema:ref:`sec-DynamicTables`, which are used throughout the NWB schema to store information about time intervals,
+electrodes, or spiking output, provide the ability to dynamically add columns without the need for extensions, and can
+help avoid the need for custom extensions in many cases.
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]: I prefer the original text. I don't like the use of "absolutely" here. It sounds like we are discouraging users from creating extensions

File: docs/best_practices/extensions.rst [hunk 3]
```diff
@@ -1,39 +1,49 @@
...
+
+If an extension is ultimately required, tutorials for the process may be found through the
+:nwb-overview:`NWB Overview for Extensions <extensions_tutorial/extensions_tutorial_home.html>`.
+
+It is also encouraged for extensions to self-contain their own check functions for Best Practices defined for those
```
Comment (bendichter) [inline docs/best_practices/extensions.rst:17]:
```suggestion
It is also encouraged for extensions to contain their own check functions for Best Practices defined for those
```

File: docs/best_practices/extensions.rst [hunk 4]
```diff
@@ -1,39 +1,49 @@
...
+particular extensions. See the NWBInspector Developer instructions for how to do this.  # TODO find link for that
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
It is also encouraged for extensions to contain their own check functions for their own best practices. See the NWBInspector Developer instructions for how to do this.  # TODO find link for that
```

File: docs/best_practices/extensions.rst [hunk 5]
```diff
@@ -1,39 +1,49 @@
...
 
 
+
 Provide Documentation
 ~~~~~~~~~~~~~~~~~~~~~
 
-When creating extensions be sure to provide meaningful documentation as part of the extension specification, of all
-fields (groups, datasets, attributes, links etc.) to describe what they store and how they are used.
+When creating extensions be sure to provide thorough, meaningful documentation as part of the extension specification.
+Be sure to explain all fields (groups, datasets, attributes, links etc.) and describe what they store and how they
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
Explain all fields (groups, datasets, attributes, links etc.) and describe what they store and how they
```

File: docs/best_practices/extensions.rst [hunk 6]
```diff
@@ -1,39 +1,49 @@
...
 
 
 Write the Specification to the NWBFile
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When using pynwb, you can store the specification (core and extension)  within the NWBFile by using
-. Caching the specification is preferable, particularly if you are using a
-custom extension, because this ensures that anybody who receives the data also receives the necessary data to
-interpret it.
+When using pynwb, you can store the specification (core and extension) within the NWBFile. Caching the specification is
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
When using pynwb, you can store the specification (core and extension) within the NWBFile through caching. Caching the specification is
```

File: docs/best_practices/tables.rst [hunk 7]
```diff
@@ -1,36 +1,49 @@
 Tables
 ======
 
-The DynamicTable data type that NWB uses allows you to define custom columns, which offer a high degree of flexibility.
+The :nwb-schema:ref:`dynamictable` data type that NWB uses allows you to define custom columns, which offer a high
```
Comment (bendichter) [inline docs/best_practices/tables.rst]:
```suggestion
The :nwb-schema:ref:`dynamictable` data type allows you to define custom columns, which offer a high
```

File: docs/best_practices/tables.rst [hunk 8]
```diff
@@ -1,36 +1,49 @@
...
 
-Times are always stored in seconds in NWB. This rule applies to times in TimeSeries, TimeIntervals and across NWB:N in general. E.g., in TimeInterval
-objects such as the trials and epochs table, start_time and stop_time should both be in seconds with respect to the timestamps_reference_time (which by
-default is set to the session_start_time).
-Additional time columns in TimeInterval tables (e.g., trials) should have _time as name suffix. E.g., if you add more times in the trials table, for
-instance a subject response time, name it with _time at the end (e.g. response_time) and store the time values in seconds from the timestamps_reference_time,
-just like start_time and stop_time.
+Use boolean values where appropriate. Although boolean values (``True``/``False``) are not used in the core schema,
+they are a supported data type, and we encourage the use of :nwb-schema:ref:`dynamictable` columns with boolean
+values. For instance, boolean values would be appropriate for a correct custom column to the trials table.
```
Comment (bendichter) [inline docs/best_practices/tables.rst]: We need more here. We'll need to explain how these values are written in HDF5.

File: docs/best_practices/tables.rst [hunk 9]
```diff
@@ -1,36 +1,49 @@
...
+
+
+Timing Columns
+~~~~~~~~~~~~~~
+
+Times are always stored in seconds in NWB. This rule applies to times in :nwb-schema:ref:`sec-TimeSeries`,
+:nwb-schema:ref:`sec-TimeIntervals` and across NWB in general. *E.g.*, in :nwb-schema:ref:`sec-TimeIntervals`
+objects such as the :nwb-schema:ref:`TrialsTable <sec-groups-intervals-trials>` and
+:nwb-schema:ref:`EpochTable <epochs>`, ``start_time`` and ``stop_time`` should both be in seconds with respect to the
+``timestamps_reference_time`` of the :nwb-schema:ref:`sec-NWBFile` (which by default is also set to the
```
Comment (bendichter) [inline docs/best_practices/tables.rst]:
```suggestion
``timestamps_reference_time`` of the :nwb-schema:ref:`sec-NWBFile` (which by default is the
```

File: docs/best_practices/tables.rst [hunk 10]
```diff
@@ -1,36 +1,49 @@
...
+``session_start_time``, see :ref:`best_practice_global_time_reference` for more details).
+
+Additional time columns in :nwb-schema:ref:`sec-TimeIntervals` tables, such as the
+:nwb-schema:ref:`TrialsTable <sec-groups-intervals-trials>` should have ``_time`` appended as a suffix to the name.
+*E.g.*, if you add more times in the :nwb-schema:ref:`TrialsTable <sec-groups-intervals-trials>`, such as a subject
```
Comment (bendichter) [inline docs/best_practices/tables.rst]: TrialsTable?

File: docs/best_practices/tables.rst [hunk 11]
```diff
@@ -1,36 +1,49 @@
...
+response time, name it ``response_time`` and store the time values in seconds from the ``timestamps_reference_time``
+of the :nwb-schema:ref:`sec-NWBFile`, just like ``start_time`` and ``stop_time``.
```
Comment (bendichter) [inline docs/best_practices/tables.rst:96]:
```suggestion
of the :nwb-schema:ref:`sec-NWBFile`, just like ``start_time`` and ``stop_time``.
This convention is used by downstream processing tools. For instance, NWBWidgets uses these times to create peri-stimulus time histograms relating spiking activity to trial events.
```

File: docs/conf_extlinks.py [hunk 12]
```diff
@@ -60,4 +60,5 @@
     "nwb-schema": ("https://nwb-schema.readthedocs.io/en/latest/", None),
     # "pandas": ("https://pandas.pydata.org/pandas-docs/stable/", None),
     # "nct": ("https://nwb-conversion-tools.readthedocs.io/", None),
+    "nwb-schema": ("https://nwb-schema.readthedocs.io/en/latest/", None),
```
Comment (bendichter) [inline docs/conf_extlinks.py]: It looks like this is now duplicated
```suggestion
```

File: docs/best_practices/extensions.rst [hunk 13]
```diff
@@ -1,40 +1,50 @@
...
 ==========
 
-Extend only when necessary. Extensions are an essential mechanism to integrate data with NWB that is otherwise not
-supported. However, we here need to consider that there are certain costs associated with extensions, e.g., cost of
-creating, supporting, documenting, and maintaining new extensions and effort for users to use and learn extensions.
-As such, users should attempt to use core neurodata_types or existing extentions before creating extensions.
-``DynamicTables``, which are used throughout the NWB schema e.g., to store information about time intervals and
-electrodes, provide the ability to dynamically add columns without the need for extensions, and can help avoid the
-need for custom extensions in many cases.
+Extend existing NWBFile structures only when necessary. Extensions are an essential mechanism to integrate
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
Extend the core NWB schema only when necessary. Extensions are an essential mechanism to integrate
```

File: docs/best_practices/extensions.rst [hunk 14]
```diff
@@ -1,40 +1,50 @@
...
-As such, users should attempt to use core neurodata_types or existing extentions before creating extensions.
-``DynamicTables``, which are used throughout the NWB schema e.g., to store information about time intervals and
-electrodes, provide the ability to dynamically add columns without the need for extensions, and can help avoid the
-need for custom extensions in many cases.
+Extend the core NWB schema only when necessary. Extensions are an essential mechanism to integrate
+data with NWB that is otherwise not supported. However, we here need to consider that there are certain costs associated
+with extensions, *e.g.*, cost of creating, supporting, documenting, and maintaining new extensions and effort for users
+to use and learn already-created extensions.
+
+As such, users should attempt to use core ``neurodata_types`` or pre-existing extentions before creating new ones.
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
to use and learn already-created extensions. As such, users should attempt to use core ``neurodata_types`` or pre-existing extentions before creating new ones.
```

File: docs/best_practices/extensions.rst [hunk 15]
```diff
@@ -1,40 +1,48 @@
...
-need for custom extensions in many cases.
+Extend the core NWB schema only when necessary. Extensions are an essential mechanism to integrate
+data with NWB that is otherwise not supported. However, we here need to consider that there are certain costs associated
+with extensions, *e.g.*, cost of creating, supporting, documenting, and maintaining new extensions and effort for users
+to use and learn already-created extensions. As such, users should attempt to use core ``neurodata_types`` or pre-existing extentions before creating new ones.
+:nwb-schema:ref:`sec-DynamicTables`, which are used throughout the NWB schema to store information about time intervals,
+electrodes, or spiking output, provide the ability to dynamically add columns without the need for extensions, and can
+help avoid the need for custom extensions in many cases.
+
+If an extension is ultimately required, tutorials for the process may be found through the
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
If an extension is required, tutorials for the process may be found through the
```

File: docs/best_practices/extensions.rst [hunk 16]
```diff
@@ -1,40 +1,48 @@
...
 
 
 Write the Specification to the NWBFile
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-When using pynwb, you can store the specification (core and extension)  within the NWBFile by using
-. Caching the specification is preferable, particularly if you are using a
-custom extension, because this ensures that anybody who receives the data also receives the necessary data to
-interpret it.
+When using :pynwb-docs:`PyNWB <>`, you can store the specification (core and extension) within the NWBFile through caching.
```
Comment (bendichter) [inline docs/best_practices/extensions.rst]:
```suggestion
You can store the specification (core and extension) within the NWBFile through caching.
```

File: docs/best_practices/tables.rst [hunk 17]
```diff
@@ -1,7 +1,8 @@
 Tables
 ======
 
-The DynamicTable data type that NWB uses allows you to define custom columns, which offer a high degree of flexibility.
+The :nwb-schema:ref:`dynamictable` data type allows you to define custom columns, which offer a high
```
Comment (bendichter) [inline docs/best_practices/tables.rst]:
```suggestion
The :nwb-schema:ref:`dynamictable` data type stores tabular data. It also allows you to define custom columns, which offer a high
```

**Pull request merged**